core/src/util/mod.rs

```diff
@@ -79,3 +94,30 @@ macro_rules! set {
             ].into_iter())
         };
     }
+
+#[cfg(test)]
+mod filtering {
+    use super::{is_match, to_name_regex};
+
+    #[test]
+    fn test_to_name_regex() {
+        assert!(to_name_regex("%").is_none());
+        assert!(to_name_regex("").is_none());
+        assert!(to_name_regex("filter").is_some());
+        assert!(to_name_regex("customers").is_some());
+    }
+
+    #[test]
+    fn test_is_match() {
+        assert!(is_match("filter", &to_name_regex("%")));
+        assert!(is_match("filter", &to_name_regex("filter")));
+        assert!(is_match("downtimes", &to_name_regex("downtimes")));
+        assert!(is_match("status", &to_name_regex("status")));
+        assert!(is_match("customer_sales", &to_name_regex("customer_sales")));
+        assert!(is_match("field_name", &to_name_regex("field_name")));
```

Comment:
It might be overkill, but maybe there could be a test case where the filter includes `%` just to make sure we hit both `replace(...)` pieces?